framework/assets/yii.activeForm.js

```diff
@@ -565,6 +565,70 @@
         }, validationDelay ? validationDelay : 200);
     };
 
+    /**
+     * Compares two value whatever it objects, arrays or simple types
+     * @param val1
+     * @param val2
+     * @returns boolean
+     */
+    var isEqual = function(val1, val2) {
+        // objects
+        if (val1 instanceof Object) {
+            return isObjectsEqual(val1, val2)
+        }
+
+        // arrays
+        if (Array.isArray(val1)) {
```

Comment:
Arrays are objects as well. Is special case necessary?

Reply:
Yes, we do need the special case due to different approach to compare them.